pypopquiz/sources.py

```diff
@@ -29,9 +30,17 @@ def get_source(source_data: Dict[str, Any], output_dir: Path, input_dir: Path, w
 
     if source_type == "youtube":
         ppq.io.log("Downloading video '{:s}' from Youtube...".format(source_url))
-        video = YouTube("https://www.youtube.com/watch?v={:s}".format(source_url))
-        video = video.streams.filter(subtype=source_data["format"]).first()
-        video.download(output_path=str(output_dir / ppq.io.SOURCES_BASE_FOLDER), filename=source_url)
+
+        if downloader == 'pytube':
+            video = YouTube("https://www.youtube.com/watch?v={:s}".format(source_url))
+            video = video.streams.filter(subtype=source_data["format"]).first()
+            video.download(output_path=str(output_dir / ppq.io.SOURCES_BASE_FOLDER), filename=source_url)
+        elif downloader == 'youtube-dl':
+            output_tpl = str(output_dir / ppq.io.SOURCES_BASE_FOLDER) + "/%(id)s.%(ext)s"
```

Comment:
What is this % syntax for string formatting? But anyway isn't this file name already set somewhere?

Reply:
By default youtube-dl figures out the title of the video from the data it gets from youtube, and it uses it as part of its output file name. To make the output compatible with pytube, I tell it to format the filename as "id.ext". youtube-dl takes the literal string `%(id)s.%(ext)s` and uses it to format the output that it creates.